File: hw/bonilla/U3/HW23Strategy/Strategy/src/main/java/ec/edu/espe/strategy/model/SortingContext.java
package ec.edu.espe.strategy.model;

import ec.edu.espe.strategy.controller.BubbleSort;
import ec.edu.espe.strategy.controller.InsertionSort;
import ec.edu.espe.strategy.controller.QuickSort;
import ec.edu.espe.strategy.controller.SortingStrategy;

/**
 *
 * @author Bonilla David SoftCrafters
 */
public class SortingContext {

    private SortingStrategy ss;

    public int[] sort(int data[]) {
        int size = data.length;
        ss = setSortStrategy(size);
        return ss.sort(data);
    }

    public SortingStrategy setSortStrategy(int n) {
        if (n > 0 && n < 30) {
            ss = new BubbleSort();
        }

        if (n >= 30 && n < 100) {
            ss = new InsertionSort();
        }

        if (n >= 100) {
            ss = new QuickSort();
        }

        return ss;
    }
}

File: hw/yanez/u3/HW22Strategy/strategy_sorting/controller/SortingContext.py
from model.BubbleSort import BubbleSort
from model.InsertionSort import InsertionSort
from model.QuickSort import QuickSort

class SortingContext:
    def __init__(self):
        self.strategy = None

    def sort(self, data: list) -> list:
        size = len(data)
        self.set_sort_strategy(size)
        return self.strategy.sort(data)

    def set_sort_strategy(self, size: int):
        if 0 < size < 30:
            self.strategy = BubbleSort()
        elif 30 <= size < 100:
            self.strategy = InsertionSort()
        else:
            self.strategy = QuickSort()